src/main/kotlin/app/revanced/patches/tumblr/featureflags/patch/OverrideFeatureFlagsPatch.kt

```diff
@@ -0,0 +1,118 @@
+import app.revanced.patcher.annotation.Package
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructionsWithLabels
+import app.revanced.patcher.extensions.or
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.patches.tumblr.featureflags.fingerprints.GetFeatureValueFingerprint
+import com.android.tools.smali.dexlib2.AccessFlags
+import com.android.tools.smali.dexlib2.builder.MutableMethodImplementation
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethod
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethodParameter
+
+@Name("Override feature flags")
+@Description("Forcibly set the value of A/B testing features of your choice.")
+@Compatibility([Package("com.tumblr")])
+class OverrideFeatureFlagsPatch : BytecodePatch(
+    listOf(GetFeatureValueFingerprint)
+) {
+    override fun execute(context: BytecodeContext) = GetFeatureValueFingerprint.result?.let {
+        // The method we want to inject into does not have enough registers,
+        // so instead of dealing with increasing the register count, we add a
+        // helper method "getValueOverride" to the class and pass it the feature object.
+        // If the helper returns null, we let the function run normally, otherwise we return the helper result value.
+        val helperMethod = ImmutableMethod(
+            it.method.definingClass,
+            "getValueOverride",
+            listOf(ImmutableMethodParameter("Lcom/tumblr/configuration/Feature;", null, "feature")),
+            "Ljava/lang/String;",
+            AccessFlags.PUBLIC or AccessFlags.FINAL,
+            null,
+            null,
+            MutableMethodImplementation(4)
+        ).toMutable().apply {
+            // At the start of the helper, we get the String representation of the enum once.
+            // At the end of the helper, we return null.
+            // Between these two, we will later insert the checks & returns for the overrides
+            addInstructions(
+                0,
+                """
+                    # toString() the enum value
+                    invoke-virtual {p1}, Lcom/tumblr/configuration/Feature;->toString()Ljava/lang/String;
+                    move-result-object v0
+                    
+                    # !!! If you add more instructions above this line, update helperInsertIndex below!
+                    # Here we will insert one compare & return for every registered Feature override
+                    # This is done below in the addOverride function
+                    
+                    # If none of the overrides returned a value, we should return null
+                    const/4 v0, 0x0
+                    return-object v0
+                """
+            )
```

Comment:
It is suggested to move such implementations to [ReVanced Integrations](https://github.com/revanced/revanced-integrations). How it works is that you implement your method there, ReVanced Patcher merges the compiled method and you can use it in the patches. If the method references code from the app itself, you can add classes to the "dummy" module. Since your patch, later on, adds instructions to that helper method, you can use a fingerprint to locate the method you added to ReVanced Integrations.

Reply:
> such implementations

What makes this an implementation to move over there? Simply because it's a new method?  
It seems a bit silly to separate it into the integrations repo since it's just a toString call and a return, but if you say so then I'll do it this way.

Reply:
Basically readability. It is much easier to maintain and read a java method instead of bytecode.

Reply:
(as discussed in discord) moving the function into integrations doesn't really work since it would compile away the unused variable and get complied with only one register. we need at least two registers for the later injected instructions, which just moves the problem of too few registers which was the reason for creating the helper function in the first place.  
I've updated comments to also explain this and to provide the java-equivalent of the injected smali instructions.